packages/pigeon/mock_handler_tester/test/message.dart

```diff
@@ -1,173 +1,148 @@
+      ..error = pigeonMap['error'] as String;
   }
 }
 
 class SearchRequest {
   String query;
   int anInt;
   bool aBool;
+
   // ignore: unused_element
-  Map<dynamic, dynamic> _toMap() {
-    final Map<dynamic, dynamic> pigeonMap = <dynamic, dynamic>{};
+  Object encode() {
+    final Map<Object, Object> pigeonMap = <Object, Object>{};
     pigeonMap['query'] = query;
     pigeonMap['anInt'] = anInt;
     pigeonMap['aBool'] = aBool;
     return pigeonMap;
   }
 
   // ignore: unused_element
-  static SearchRequest _fromMap(Map<dynamic, dynamic> pigeonMap) {
-    if (pigeonMap == null) {
-      return null;
-    }
-    final SearchRequest result = SearchRequest();
-    result.query = pigeonMap['query'];
-    result.anInt = pigeonMap['anInt'];
-    result.aBool = pigeonMap['aBool'];
-    return result;
+  static SearchRequest decode(Object message) {
+    final Map<Object, Object> pigeonMap = message as Map<Object, Object>;
+    return SearchRequest()
+      ..query = pigeonMap['query'] as String
+      ..anInt = pigeonMap['anInt'] as int
+      ..aBool = pigeonMap['aBool'] as bool;
   }
 }
 
 class Nested {
   SearchRequest request;
+
   // ignore: unused_element
-  Map<dynamic, dynamic> _toMap() {
-    final Map<dynamic, dynamic> pigeonMap = <dynamic, dynamic>{};
-    pigeonMap['request'] = request == null ? null : request._toMap();
+  Object encode() {
+    final Map<Object, Object> pigeonMap = <Object, Object>{};
+    pigeonMap['request'] = request == null ? null : request.encode();
     return pigeonMap;
   }
 
   // ignore: unused_element
-  static Nested _fromMap(Map<dynamic, dynamic> pigeonMap) {
-    if (pigeonMap == null) {
-      return null;
-    }
-    final Nested result = Nested();
-    result.request = SearchRequest._fromMap(pigeonMap['request']);
-    return result;
+  static Nested decode(Object message) {
+    final Map<Object, Object> pigeonMap = message as Map<Object, Object>;
+    return Nested()
+      ..request = pigeonMap['request'] != null
+          ? SearchRequest.decode(pigeonMap['request'])
+          : null;
   }
 }
 
 abstract class FlutterSearchApi {
   SearchReply search(SearchRequest arg);
   static void setup(FlutterSearchApi api) {
     {
-      const BasicMessageChannel<dynamic> channel = BasicMessageChannel<dynamic>(
+      const BasicMessageChannel<Object> channel = BasicMessageChannel<Object>(
           'dev.flutter.pigeon.FlutterSearchApi.search', StandardMessageCodec());
-      channel.setMessageHandler((dynamic message) async {
-        final Map<dynamic, dynamic> mapMessage =
-            message as Map<dynamic, dynamic>;
-        final SearchRequest input = SearchRequest._fromMap(mapMessage);
-        final SearchReply output = api.search(input);
-        return output._toMap();
-      });
+      if (api == null) {
+        channel.setMessageHandler(null);
+      } else {
+        channel.setMessageHandler((Object message) async {
+          if (message == null) {
```

Comment:
We shouldn't bypass calling into the api.  Either we support null here and pass it through to the api, or we don't and we assert.  Silently bypassing the API is just going to create hard to debug situations.

Reply:
But then FlutterSearchApi.search has to take a nullable argument. Do we want that?

Reply:
I think this is the main open question here.

Reply:
Ok sounds like from the previous comment that you're good leaving this as-is.